### PR TITLE
React/DP-15689: Consolidate teaser & Add Data Teasers

### PR DESCRIPTION
**Pull request by avrilpearl**
# Minor

## Added
- (React) [Icon] DP-15689: Added `mail` and `calendar` svg for Icon options. #766
- (React) [PhoneNumber] DP-15689: Added PhoneNumber component to library. #766
- (React) [EventTime] DP-15689: Added EventTime component to library. #766
- (React) [Email] DP-15689: Added Email component to library. #766
- (React) [Address] DP-15689: Added Address component to library. #766
- (React) [GenTeaser] DP-15689: Added GenTeaser component to library. #766
- (React) [Dependency] DP-15689: Added react-html-parser as dependency. #766

## Changed
- (React) [Dependency] DP-15689: Bumps eslint-plugin-react from 7.11.1 to 7.14.3. #766
- (React) [ButtonWithIcon] DP-15689: Added `size` prop to component. #766
- (React) [DecorativeLink] DP-15689: Added `icon` and `details` props to component. #766
- (React) [PressTeaser, GeneralTeaser, Teaser] DP-15689: Flagged components for removal & archiving in version 10. Removed story from storybook. #766

### CODE REVIEW

**Comment by clairesunstudio** [comment]
Great job consolidating and expanding the teaser component, can't wait to use it! 

Some quick design feedback before diving in more and doing code review. I noticed you downsized the fonts in some places. 
For the actionable link items, e.g. the area of the links are too small for users to act on accurately. I suggest we keep them the same at 1.2rem and leave a 8px margin-bottom. (Just want to mention, I like the current organization of the teaser stories, it makes it so easy to review)
Screen Shot 2019-10-07 at 12 55 52 PM

File: assets/scss/02-molecules/_link-dropdown.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@
   display: inline-block;
 
   .ma__link-dropdown-menu {
-    position: relative;
+    position: fixed;
```
Comment (clairesunstudio) [inline assets/scss/02-molecules/_link-dropdown.scss:6]: is "link-dropdown" created for this specific use only?

File: react/src/components/atoms/buttons/ButtonWithIcon/index.js [hunk 2]
```diff
@@ -44,6 +45,8 @@ ButtonWithIcon.propTypes = {
   text: PropTypes.string,
   /** HTML <button> 'type' attribute  */
   type: PropTypes.oneOf(['submit', 'reset', 'button', '']),
+  /** Create a smaller button */
+  size: PropTypes.oneOf(['', 'small', 'large']),
```
Comment (clairesunstudio) [inline react/src/components/atoms/buttons/ButtonWithIcon/index.js:49]: update story knob

File: changelogs/DP-15689.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- (React) [Email] DP-15689: Added Email component to library. #766
+- (React) [Address] DP-15689: Added Address component to library. #766
+- (React) [GenTeaser] DP-15689: Added GenTeaser component to library. #766
+- (React) [Dependency] DP-15689: Added react-html-parser as dependency. #766
+
+## Changed
+- (React) [Dependency] DP-15689: Bumps eslint-plugin-react from 7.11.1 to 7.14.3. #766
+- (React) [ButtonWithIcon] DP-15689: Added `size` prop to component. #766
+- (React) [DecorativeLink] DP-15689: Added `icon` and `details` props to component. #766
+- (React) [PressTeaser, GeneralTeaser, Teaser] DP-15689: Flagged components for removal & archiving in version 10. Removed story from storybook. #766
```
Comment (clairesunstudio) [inline changelogs/DP-15689.md]:
```suggestion

## Removed
- (React) [PressTeaser, GeneralTeaser, Teaser] DP-15689: Flagged components for removal & archiving in version 10. Removed story from storybook. #766
```

File: react/src/components/organisms/GenTeaser/TeaserSearch/index.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    const { placeholder, id, ...rest } = this.props;
+    return(
+      <HeaderSearch
+        buttonSearch={{
+          'aria-label': '',
+          onClick: (e) => this.onClick(e),
+          text: 'Search',
+          usage: ''
+        }}
+        defaultText=""
```
Comment (clairesunstudio) [inline react/src/components/organisms/GenTeaser/TeaserSearch/index.js]: I wonder if we can just pass `queryInput` into `defaultText` as this prop is used and tracked similarly in HeaderSearch

File: react/src/components/organisms/GenTeaser/utils.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      calendarUrl += '&path=/calendar/view/Month';
+      break;
+
+    default:
+      calendarUrl = [
+        'BEGIN:VCALENDAR',
+        'VERSION:2.0',
+        'BEGIN:VEVENT',
+        `URL:${document.URL}`,
+        `DTSTART:${formatTime(event.startTime)}`,
```
Comment (clairesunstudio) [inline react/src/components/organisms/GenTeaser/utils.js]: The startDate and endDate don't seem to reflect what's passed in in the downloaded file, seems to be adding today's date instead.

Comment (clairesunstudio) [reply react/src/components/organisms/GenTeaser/utils.js]: Same with outlook and google cal

Comment (clairesunstudio) [reply react/src/components/organisms/GenTeaser/utils.js]: Screen Shot 2019-10-09 at 4 19 36 PM
Screen Shot 2019-10-09 at 4 19 57 PM

File: react/src/components/organisms/GenTeaser/TeaserSearch/index.js (hunk 4 above)
Comment (clairesunstudio) [reply react/src/components/organisms/GenTeaser/TeaserSearch/index.js]: updated in 506913d and remove unnecessary state tracking

File: react/src/components/organisms/GenTeaser/utils.js (hunk 5 above)
Comment (avrilpearl) [reply react/src/components/organisms/GenTeaser/utils.js]: @clairesunstudio --> fixed this just had the wrong variable called out in the utils function.

Comment (clairesunstudio) [reply react/src/components/organisms/GenTeaser/utils.js]: confirmed fixed!

**Comment by clairesunstudio** [comment]
Changes made:
- reorganized doc and code chunk for GenTeaser to reflect the order or rendering 
- changed the html order/structure for moreInfo (event and address)
- refactored and removed unnecessary state tracking in TeaserSearch
- refactored and update css styling
- change "Add to Calendar" button to quarternary instead of quaternary simple

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**